### ISSUE TITLE
Bech32m unavailable for encrypted wallets

### ISSUE THREAD

**Issue by hebasto** [bitcoin-core/gui#571]
bitcoin-core/gui#459 introduced an ability to chose Bech32m address type in the GUI.

On master and on v23.0rc2 this option is unavailable for a just created encrypted wallet, at least on Ubuntu 22.04.

For macOS a slightly different behavior [reported](https://github.com/bitcoin/bitcoin/issues/24501#issuecomment-1081621502).

See details:
- https://github.com/bitcoin/bitcoin/issues/24501#issuecomment-1080821599
- https://github.com/bitcoin/bitcoin/issues/24501#issuecomment-1081598045

**Comment by hebasto** [comment]
cc @Sjors @promag @achow101

**Comment by Sjors** [comment]
I'm able to reproduce. For some reason _in the GUI_ we're not producing a taproot descriptor when encryption is enabled (see `listdescriptors`). This does _not_ happen for wallets created via the `createwallet` RPC.

Thanks for catching that @eriknylund.

**Comment by Sjors** [comment]
Uh no never mind, it _does_ generate the descriptor. I selected the wrong wallet in the console. Pfew. So it's a pure UI issue, investigating...

**Comment by eriknylund** [comment]
> Uh no never mind, it _does_ generate the descriptor. I selected the wrong wallet in the console. Pfew. So it's a pure UI issue, investigating...
> 
> Restarting Bitcoin Core does add Bech32m to the menu. Maybe some reference to the receive screen got borked in the passphrase dialogs? See #509.
> 
> @eriknylund if you're _not_ seeing Bech32m after a restart, are you sure you checked the "Descriptor" box (it's selected by default)? You can run `getwalletinfo` in the Console (make sure to select the correct wallet in the console too) and look if you see `"descriptors": true`. Only descriptor wallets have Taproot support.

Something sure looks borked. I cannot even generate receive addresses on the other three types, the `Create new receiving address` button is disabled.
![image](https://user-images.githubusercontent.com/20860124/160646661-d066ddc2-aa70-4f2f-9412-83f71cdd5d06.png)
This is the output from the Console:
```
{
  "walletname": "Mainnet Wallet 1 Encrypted",
  "walletversion": 169900,
  "format": "sqlite",
  "balance": 0.00000000,
  "unconfirmed_balance": 0.00000000,
  "immature_balance": 0.00000000,
  "txcount": 0,
  "keypoolsize": 0,
  "keypoolsize_hd_internal": 0,
  "unlocked_until": 0,
  "paytxfee": 0.00000000,
  "private_keys_enabled": true,
  "avoid_reuse": false,
  "scanning": false,
  "descriptors": true,
  "external_signer": false
}
```

**Comment by Sjors** [comment]
@eriknylund this last screenshot is from after a restart?

**Comment by eriknylund** [comment]
> @eriknylund this last screenshot is from after a restart?

@Sjors Correct.

**Comment by hebasto** [comment]
A fixed has been suggested in bitcoin/bitcoin#24711.
